lib/BabelPluginExportToFunction.js

```diff
@@ -37,11 +50,14 @@ function BabelPluginExportToFunction(configAst) {
 
                 if (isDefaultExport(left) && isIdentifier(right)) {
                     const functionIdentifier = path.node.right;
-                    const binding = path.scope.bindings[functionIdentifier.name];
-
-                    if (binding && isValidFunctionExpression(binding.path.node.init)) {
-                        const functionDefinition = binding.path.node.init;
+                    const node = recursivelyResolveBinding(path.scope.bindings, functionIdentifier);
+                    const functionDefinition = R.cond([
+                        [ isValidFunctionDeclaration, R.identity ],
+                        [ R.pipe(R.propOr({}, 'init'), isValidFunctionExpression), R.prop('init') ],
```

Comment:
Maybe out of scope of this PR, but does `isValidFunctionExpression` support arrow functions?